packages/rsocket-graphql-apollo-server/src/ApolloServer.ts

```diff
@@ -0,0 +1,170 @@
+import { defer, from, Observable, of, switchMap } from "rxjs";
+import { ExecutionResult } from "graphql/execution/execute";
+
+export interface RSocketContext {
+  payload: Payload;
+}
+
+export class ApolloServer<
+  ContextFunctionParams = RSocketContext
+> extends ApolloServerBase<ContextFunctionParams> {
+  private readonly schema: GraphQLSchema;
+
+  constructor(config: Config<ContextFunctionParams>) {
+    super(config);
+    this.schema = makeExecutableSchema({
+      typeDefs: config.typeDefs,
+      resolvers: config.resolvers,
+    });
+  }
+
+  async createGraphQLServerOptions(payload: Payload): Promise<GraphQLOptions> {
+    const contextParams: RSocketContext = { payload };
+    return this.graphQLServerOptions(contextParams);
+  }
+
+  public getHandler(): Partial<RSocket> {
+    return {
+      // handle single Query/Mutation
+      requestResponse: (
+        payload: Payload,
+        responderStream: OnTerminalSubscriber &
+          OnNextSubscriber &
+          OnExtensionSubscriber
+      ): Cancellable & OnExtensionSubscriber => {
+        const subscription = this.runQueryOperation(payload).subscribe(
+          this.queryOperationSubscriber(responderStream)
+        );
+
+        return {
+          cancel(): void {
+            subscription.unsubscribe();
+          },
+          onExtension(): void {},
+        };
+      },
+
+      // handle Subscriptions
+      requestStream: (
+        payload: Payload,
+        initialRequestN: number,
+        responderStream: OnTerminalSubscriber &
+          OnNextSubscriber &
+          OnExtensionSubscriber
+      ): Requestable & Cancellable & OnExtensionSubscriber => {
+        const subscription = this.runSubscriptionOperation(payload).subscribe(
+          this.subscriptionOperationSubscriber(responderStream)
+        );
+
+        return {
+          cancel(): void {
+            subscription.unsubscribe();
+          },
+          onExtension(): void {},
+          request(requestN: number): void {},
+        };
+      },
+    };
+  }
+
+  private queryOperationSubscriber(
+    responderStream: OnTerminalSubscriber &
+      OnNextSubscriber &
+      OnExtensionSubscriber
+  ) {
+    return {
+      next({ graphqlResponse }) {
+        responderStream.onNext(
+          {
+            data: Buffer.from(graphqlResponse),
+          },
+          true
+        );
+      },
+      error(e) {
+        responderStream.onError(e);
+      },
+    };
+  }
+
+  private runQueryOperation(payload: Payload): Observable<any> {
+    const query = parsePayloadForQuery(payload);
+
+    return defer(() =>
+      from(
+        runHttpQuery([], {
+          method: "POST",
+          options: () => {
+            return this.createGraphQLServerOptions(payload);
+          },
+          query,
+          request: new Request("/graphql", {
+            headers: new Headers(),
+            method: "POST",
+          }),
+        })
```

Comment:
From what I can tell, `runHttpQuery` seems to be the best entry point into Apollo's logic for handling a query. This feels strange for our use case, seeing as we are not HTTP, so I'll likely reach out to Apollo and see if there is a better way to process a query that is completely divorced from HTTP concepts.